server/session/handler_crafting.go

```diff
@@ -238,3 +225,8 @@ func grow(i recipe.Item, count int) recipe.Item {
 	}
 	panic(fmt.Errorf("unexpected recipe item %T", i))
 }
+
+// noinspection ALL
+//
+//go:linkname duplicateStack github.com/df-mc/dragonfly/server/internal/iteminternal.DuplicateStack
+func duplicateStack(stack item.Stack, item world.Item) item.Stack
```

Comment:
Is this necessary? If so, might be a good idea to move this function elsewhere after all to avoid having to use compiler directives. The fewer of these the better.

Reply:
I've pushed an `item.Stack.WithItem()` method in the latest commit that implements this. If you use that for crossbows and remove this function, you should be set.